src/hooks/useWritingProblems.ts

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useState } from "react";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { useDispatch } from "react-redux";
+import { HttpMethod, RootStackParamList } from "../types";
+import getProblem from "../assets/prompts";
+
+import gameDescriptions from "../screens/Stacks/gameDescriptions";
+import { completedWriting } from "../redux/reducers/gameDetailsReducer";
+import { internalRequest } from "../requests";
+
+type Props = NativeStackScreenProps<RootStackParamList, "WritingMain">;
+
+const TOTAL_TIME = gameDescriptions.Writing.minutes * 60;
+export default function useWritingProblems({ navigation, route }: Props) {
+  const dispatch = useDispatch();
+  const [problem, setProblem] = useState(getProblem());
+  const [questionsAttempted, setQuestionsAttempted] = useState(0);
+  const [questionsCorrect, setQuestionsCorrect] = useState(0);
+
+  const updateStatsOnAnswer = useCallback((isCorrect: boolean) => {
+    setQuestionsAttempted((prev) => prev + 1);
+    if (isCorrect) {
+      setQuestionsCorrect((prev) => prev + 1);
+    }
+  }, []);
+
+  const getNewProblem = useCallback(() => {
+    setProblem(getProblem());
+  }, []);
+
+  const onTimeComplete = useCallback(() => {
+    const average =
+      questionsAttempted > 0 ? TOTAL_TIME / questionsAttempted : 0;
+    const statistics = {
+      skipped: questionsAttempted - questionsCorrect,
+      questionsAnswered: questionsCorrect,
+      timePerQuestion: average,
+    };
```

Comment:
Make appropriate changes to the `timePerQuestion` metric, similar to what I mentioned for `wordsPerMinute` and `timePerPassage` in the reading section